plugin/throttle/throttle.go

```diff
@@ -0,0 +1,124 @@
+package throttle
+
+import (
+	"context"
+	"sync/atomic"
+	"time"
+
+	"github.com/coredns/coredns/plugin/metrics"
+
+	"github.com/prometheus/client_golang/prometheus"
+
+	"github.com/coredns/coredns/plugin"
+	"github.com/miekg/dns"
+)
+
+// Throttle holds the count of queries currently in process and limits defined by configuration
+type Throttle struct {
+	workerCount  int64
+	workerLimit  int64
+	writingCount int64
+	timeout      time.Duration
+	Next         plugin.Handler
+}
+
+// ServeDNS implements the plugin.Handler interface.
+func (t *Throttle) ServeDNS(ctx context.Context, w dns.ResponseWriter, r *dns.Msg) (int, error) {
+
+	droptIt := func(server string, dropttype string, w dns.ResponseWriter) (int, error) {
```

Comment:
Are "droptIt" and "dropttype" variable name typos? e.g.
```suggestion
	dropIt := func(server string, droptype string, w dns.ResponseWriter) (int, error) {
```

Why is this function a variable, and not a normal function?

Reply:
your are right: typo.... maybe glasses missing.